gadgets/trace_tcp/ebpf/connect.h

```diff
@@ -53,27 +55,35 @@ static __always_inline int exit_tcp_connect(struct pt_regs *ctx, int ret,
 	__u64 pid_tgid = bpf_get_current_pid_tgid();
 	__u32 tid = pid_tgid;
 	struct tuple_key_t tuple = {};
-	struct gadget_process proc = {};
+	struct extended_info ei = {};
 	struct sock **skpp;
 	struct sock *sk;
+	__u32 *fd;
 
-	skpp = bpf_map_lookup_elem(&tcp_connect_sockets, &tid);
+	skpp = bpf_map_lookup_elem(&tcp_tid_sock, &tid);
 	if (!skpp)
-		return 0;
+		goto end1;
+
+	fd = bpf_map_lookup_elem(&tcp_tid_fd, &tid);
+	if (!fd)
+		goto end1;
+	ei.fd = *fd;
 
 	if (ret)
-		goto end;
+		goto end2;
 
 	sk = *skpp;
 
 	if (!fill_tuple(&tuple, sk, family))
-		goto end;
+		goto end2;
 
-	gadget_process_populate(&proc);
-	bpf_map_update_elem(&tuplepid, &tuple, &proc, 0);
+	gadget_process_populate(&ei.proc);
+	bpf_map_update_elem(&tuplepid, &tuple, &ei, 0);
 
-end:
-	bpf_map_delete_elem(&tcp_connect_sockets, &tid);
+end2:
+	bpf_map_delete_elem(&tcp_tid_sock, &tid);
+end1:
+	bpf_map_delete_elem(&tcp_tid_fd, &tid);
```

Comment:
We might need to cleanup `tcp_tid_fd` and `tcp_tid_sock` in `tracepoint/sched/sched_process_exit` as well in case the kretprobe runs above the maxactive limit with synchronous connections.